### PR TITLE
refactor initialization of client using AppConfig

### PR DESCRIPTION
**Pull request by beniwohli**
Since we dropped support for older Django versions, we can make use of
the new `AppConfig.ready` hook to initialize our handlers.

We also add automatic instrumentation of Celery tasks in the `ready`
handler.

### CODE REVIEW

File: elasticapm/contrib/django/handlers.py [hunk 1]
```diff
@@ -12,26 +12,60 @@
...
         # Fetch the request from a threadlocal variable, if available
         request = getattr(LogMiddleware.thread, 'request', None)
         request = getattr(record, 'request', request)
 
-        return super(LoggingHandler, self)._emit(record, request=request)
+        return super(LoggingHandler, self)._emit(record, request=request, **kwargs)
+
+
+def exception_handler(request=None, **kwargs):
+    def actually_do_stuff(request=None, **kwargs):
```
Comment (jalvz) [inline elasticapm/contrib/django/handlers.py]: 😏

**Comment by jalvz** [comment]
there is quite a number of implementations details I can't follow anymore (if I ever could) like the `get_client()` logic, but pretty awesome to get celery support out of the box!

tests look solid too, good stuff 💯

File: tests/contrib/django/testapp/tasks.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from ..testapp.celery import app
```
Comment (ruflin) [inline tests/contrib/django/testapp/tasks.py]: These `..` look suspicious.

Comment (beniwohli) [inline tests/contrib/django/testapp/tasks.py]: That's a relative import. But I can change to `tests.contrib.django.testapp.celery` if you prefer

Comment (ruflin) [inline tests/contrib/django/testapp/tasks.py]: All good if it works, it only thought I should probably let this double check.

**Pull request merged**